app/lib/widgets/settings/share_settings.dart

```diff
@@ -0,0 +1,235 @@
+                            labelText: 'Email',
+                          ),
+                          onSaved: (value) {
+                            _email = value ?? '';
+                          },
+                          validator: (String? value) {
+                            return (value != null &&
+                                    !EmailValidator.validate(value))
+                                ? 'Not a valid email address'
+                                : null;
+                          },
+                        ),
+                        Text("${_duration.round()} days"),
+                        CupertinoSlider(
+                          value: _duration,
+                          min: 1.0,
+                          max: 365.0,
+                          onChanged: (newDuration) {
+                            setState(() => _duration = newDuration);
+                          },
+                        ),
+                        Padding(
+                          padding: const EdgeInsets.symmetric(vertical: 6),
+                          child: ElevatedButton(
+                            onPressed: () => submitForm(Navigator.of(context)),
+                            child: const Text('Share'),
+                          ),
+                        ),
+                        error != null
+                            ? Text(
+                                error!,
+                                style: TextStyle(
+                                    color: Theme.of(context).errorColor),
+                              )
+                            : Container(),
+                        success != null
+                            ? Text(
+                                success!,
+                                style: const TextStyle(color: Colors.green),
+                              )
+                            : Container(),
+                      ],
+                    ),
+                  ],
+                ),
+              ),
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class _ShareHolderList extends State<ShareSettings> {
+  final api = obtainApiClient();
+  late Future _shareHolders;
+
+  @override
+  void initState() {
+    super.initState();
+    _shareHolders = api?.getCurrentShares(widget.bike.id);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder(
+      future: _shareHolders,
+      builder: (BuildContext context, AsyncSnapshot snapshot) =>
+          StatefulBuilder(
+        builder: (context, setState) {
+          if (snapshot.connectionState != ConnectionState.done) {
+            return Center(
+                child: Column(children: const [
+              CircularProgressIndicator(),
+              Text("Loading shares"),
+            ]));
+          }
+
+          if (snapshot.hasData) {
+            if (snapshot.data.length == 0) {
+              return const Text('No share holders');
+            } else {
```

Comment:
No need for `else` here as we return in the if statement

Reply:
✅